newsrevenuehub/rev-engine
# DEV-3521: Fix scheduled payment email header

Please fill out each section even if it's just with "N/A"

#### Plan? (if this draft/incomplete indicate what you intend to do and how)

n/a

#### What's this PR do?

- Fixes alignment issue with scheduled payment reminder emails
- Adds a debug-only route to the Django app, `/__debug_email__/recurring-contribution/` that allow previewing emails.

#### Why are we doing this? How does it help us?

- Fixes a display bug
- Makes it easier to work on styled emails

#### How should this be manually tested? Please include detailed step-by-step instructions.

To test styled emails for regressions:

1. Set up a recurring contribution to an org whose default page has a style set.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (header color and logo are correct, appear above text header) and contains accurate info.
4. Set up a recurring contribution to an org whose default page has a style set, but no header logo.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (has left-aligned NRE logo at top, no background color in header) and contains accurate info.

To test unstyled emails for regressions:

1. Set up a recurring contribution to an org whose default page doesn't have a style set.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (has left-aligned NRE logo at top) and contains accurate info.

To test free plan emails for regressions:

1. Set up a recurring contribution to an org whose default page has a style set, but whose org is on the free plan.
2. Send a reminder email for this contribution.
3. Confirm the email appearance is correct (does not use custom style, has left-aligned NRE logo at top) and contains accurate info.

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?

No.

#### Have automated unit tests been added? If not, why?

I updated existing Python tests.

#### How should this change be communicated to end users?

n/a

#### Are there any smells or added technical debt to note?

No.

#### Has this been documented? If so, where?

 I also added notes to the readme explaining how the debug email routes work.

#### What are the relevant tickets? Add a link to any relevant ones.

https://news-revenue-hub.atlassian.net/browse/DEV-3521

#### Do any changes need to be made before deployment to production (adding environment variables, for example)?

No.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

No.
